src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts

```diff
@@ -668,6 +692,20 @@ export class NotebookService extends Disposable implements INotebookService {
 		return result;
 	}
 
+
+	private _persistSoonHandle?: IDisposable;
+
+	private _persistMementos(): void {
+		this._persistSoonHandle?.dispose();
+		this._persistSoonHandle = runWhenIdle(() => {
```

Comment:
Is there a specific reason for this `runWhenIdle`? Are we expecting this data to be very large?

Reply:
good point, the data is very small and most people only have a couple of serializers, making it synchronous is likely no harm at all.